src/seedu/addressbook/data/person/Address.java

```diff
@@ -3,57 +3,164 @@
+        boolean isValidStreet = isValidStreet(personDataSplit[1]);
+        boolean isValidUnit = isValidUnit(personDataSplit[2]);
+        boolean isValidPostalcode = isValidPostalcode(personDataSplit[3]);
+        return isValidBlock && isValidStreet && isValidUnit && isValidPostalcode;
+    }
+
+    /**
+     * Returns true if a given string is a valid address PostalCode.
+     */
+    private static boolean isValidPostalcode(String string) {
+        String trimmedString = string.trim();
+        return trimmedString.trim().matches(ADDRESS_POSTALCODE_VALIDATION_REGEX);
+    }
+
+    /**
+     * Returns true if a given string is a valid address unit.
+     */
+    private static boolean isValidUnit(String string) {
+        String trimmedString = string.trim();
+        return trimmedString.matches(ADDRESS_UNIT_VALIDATION_REGEX);
     }
 
     /**
-     * Returns true if a given string is a valid person email.
+     * Returns true if a given string is a valid address street.
      */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    private static boolean isValidStreet(String string) {
+        String trimmedString = string.trim();
+        return trimmedString.matches(ADDRESS_STREET_VALIDATION_REGEX);
+
+    }
+
+    /**
+     * Returns true if a given string is a valid address block.
+     */
+    private static boolean isValidBlock(String string) {
+        String trimmedString = string.trim();
+        return trimmedString.trim().matches(ADDRESS_BLOCK_VALIDATION_REGEX);
     }
 
     @Override
     public String toString() {
-        return value;
+        return addressBlock.getBlock() + "," + addressStreet.getStreet() + "," + addressUnit.getUnit() + ","
+                + addressPostalcode.getPostalcode();
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                        && this.addressBlock.equals(((Address) other).addressBlock)
+                        && this.addressStreet.equals(((Address) other).addressStreet)
+                        && this.addressUnit.equals(((Address) other).addressUnit)
+                        && this.addressPostalcode.equals(((Address) other).addressPostalcode)); // state
+                                                                                                // check
     }
 
     @Override
     public int hashCode() {
-        return value.hashCode();
+        return this.toString().hashCode();
     }
 
+    /**
+     * Returns true if Address  is private
+     */
     public boolean isPrivate() {
-        return isPrivate;
+        return this.isPrivate;
     }
+
+    /**
+     * Returns address block object of this address
+     */
+    private AddressBlock getAddressBlock() {
```

Comment:
What is the purpose of a private getter method?
Wouldn't calling `this.addressBlock` suffice?